File: arm.64/frame/avalonia/src/find.fs
namespace Pat

open System
open System.IO

module Finder =
    let find (targetDir: string option) : string list =
        let dir =
            match targetDir with
            | Some d when not (String.IsNullOrEmpty d) -> d
            | _ ->
                let envDir = Environment.GetEnvironmentVariable "DIST_DIR"
                if not (String.IsNullOrEmpty envDir) then envDir
                else
                    let candidates = [
                        Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "../../../dist"))
                        Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "../../dist"))
                        "dist"
                        "../../dist"
                    ]
                    candidates
                    |> List.tryFind Directory.Exists
                    |> Option.defaultValue "dist"

        if not (Directory.Exists dir) then []
        else
            try
                Directory.GetFiles dir
                |> Array.filter (fun f -> Path.GetFileName f <> ".DS_Store")
                |> Array.map Path.GetFullPath
                |> Array.toList
            with _ -> []

File: arm.64/frame/blazor/src/find.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace Pat {
    public static class Finder {
        public static List<string> Find(string? targetDir = null) {
            string? dir = targetDir;
            if (string.IsNullOrEmpty(dir)) {
                string? envDir = Environment.GetEnvironmentVariable("DIST_DIR");
                if (!string.IsNullOrEmpty(envDir)) {
                    dir = envDir;
                } else {
                    string[] candidates = new string[] {
                        Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "../../../dist")),
                        Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "../../dist")),
                        "dist",
                        "../../dist"
                    };
                    dir = "dist";
                    foreach (var c in candidates) {
                        if (Directory.Exists(c)) {
                            dir = c;
                            break;
                        }
                    }
                }
            }

            var files = new List<string>();
            if (!Directory.Exists(dir)) return files;

            try {
                foreach (var file in Directory.GetFiles(dir)) {
                    if (Path.GetFileName(file) != ".DS_Store") {
                        files.Add(Path.GetFullPath(file));
                    }
                }
            } catch {}

            return files;
        }
    }
}
